saltproc/saltproc.py

```diff
@@ -55,103 +71,280 @@ def __init__(self, steps, cores, nodes, bw, exec_path, restart=False,
+
+        self.blanket_before_db = self.f.create_dataset('blanket composition before reproc',
+                                                       shape, maxshape=maxshape,
+                                                       chunks=True)
+        self.blanket_after_db = self.f.create_dataset('blanket composition after reproc',
+                                                      shape, maxshape=maxshape,
+                                                      chunks=True)
+        self.blanket_refill_tank_db = self.f.create_dataset('blanket refill tank composition',
+                                                            shape, maxshape=maxshape,
+                                                            chunks=True)
+
+        self.fissile_tank_db = self.f.create_dataset('fissile tank composition', shape, maxshape=maxshape,
+                                                     chunks=True)
+        self.waste_tank_db = self.f.create_dataset('waste tank composition',
                                                    shape, maxshape=maxshape,
                                                    chunks=True)
-        self.noble_adens_db = self.f.create_dataset('noble adensity',
-                                                    shape, maxshape=maxshape,
-                                                    chunks=True)
-        self.th_adens_db = self.f.create_dataset('Th tank adensity',
-                                                 shape, maxshape=maxshape,
-                                                 chunks=True)
-        # !! raffinate stream consider splitting by what element
-        self.rem_adens = np.zeros((5, self.number_of_isotopes))
+        # initial composition
+
         dt = h5py.special_dtype(vlen=str)
-        self.isolib_db = self.f.create_dataset('iso codes', data=self.isolib,
+        # have to encode to utf8 for hdf5 string
+        self.isolib_db = self.f.create_dataset('iso names',
+                                               data=[x.encode('utf8')
+                                                     for x in self.isoname],
                                                dtype=dt)
-
-        self.bu_adens_db_0[0, :] = self.dict_to_array(self.bumat_dict)
-        self.bu_adens_db_1[0, :] = self.dict_to_array(self.bumat_dict)
-
-        self.th232_adens_0 = self.bumat_dict['Th232']
-
-    def dict_to_array(self, bumat_dict):
-        """ Converts an OrderedDict to an array of its values
-
-        Parameters:
-        -----------
-        bumat_dict: OrderedDict
-            key: isotope name
-            value: adensity
-
-        Returns:
-        --------
-        array of bumat_dict values
+        self.isozai_db = self.f.create_dataset('iso zai', data=self.isozai)
+        # the first depleted, non-reprocessed fuel is stored in timestep 1
+        # initial composition
+        self.dep_dict = self.read_dep()
+        self.driver_before_db[0,
```

Comment:
These driver/blanket before/after lines are copy paste. They should b handled more gracefully, without copy paste ,without repeating oneself.